### ISSUE TITLE
[telescope.builtin.lsp_*]: server does not support definitionProvider

### ISSUE THREAD

**Issue by fgheng** [nvim-telescope/telescope.nvim#2898]
### Description

I have encountered the same error as this person. He mentioned this error in nvim-jdtls. The author of nvim-jdtls believes that it is an error with Telescope. I want to ask for a solution.

https://github.com/mfussenegger/nvim-jdtls/issues/589

### Neovim version

```markdown
NVIM v0.10.0-dev-2201+g226466a46
Build type: RelWithDebInfo
LuaJIT 2.1.1706185428
Run "nvim -V1 -v" for more info
```

### Operating system and version

maxOS 11.05

### Telescope version / branch / rev

main

### checkhealth telescope

```markdown
3  telescope: require("telescope.health").check()
   4 │
   5 │ Checking for required plugins
   6 │ - OK plenary installed.
   7 │ - WARNING nvim-treesitter not found.·
   8 │
   9 │ Checking external dependencies
  10 │ - OK rg: found ripgrep 13.0.0
  11 │ - OK fd: found fd 8.7.0
  12 │
  13 │ ===== Installed extensions =====
  14 │
  15 │ Telescope Extension: `fzf`
  16 │ - OK lib working as expected
  17 │ - OK file_sorter correctly configured
  18 │ - OK generic_sorter correctly configured
  19 │
  20 │ Telescope Extension: `lazy`
  21 │ - No healthcheck provided
  22 │
  23 │ Telescope Extension: `live_grep_args`
  24 │ - No healthcheck provided
  25 │
  26 │ Telescope Extension: `ui-select`
  27 │ - No healthcheck provided
```

### Steps to reproduce

LSP client configuration
```
-- Fetches Lazy if not present.
local function bootstrap_lazy()
  local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
  if not vim.loop.fs_stat(lazypath) then
    vim.fn.system({
      "git",
      "clone",
      "--filter=blob:none",
      "https://github.com/folke/lazy.nvim.git",
      "--branch=stable",
      lazypath,
    })
  end
  vim.opt.rtp:prepend(lazypath)
end
bootstrap_lazy()

require('lazy').setup({
  {
    'williamboman/mason.nvim',
    opts = {},
  },
  {
    'mfussenegger/nvim-jdtls',
    dependencies = {
      'williamboman/mason.nvim',
      'neovim/nvim-lspconfig',
      { 'nvim-telescope/telescope.nvim', dependencies = 'nvim-lua/plenary.nvim' },
    },
    config = function()
      local function setup_jdtls()
        local mason_path = require('mason-core.path')
        local function get_install_path(package)
          return require('mason-registry').get_package(package):get_install_path()
        end

        local config = {
          cmd = { mason_path.concat({ get_install_path('jdtls'), 'jdtls' }) },
        }
        require('jdtls').start_or_attach(config)
      end

      local augroup = vim.api.nvim_create_augroup('jdtls', {})
      vim.api.nvim_create_autocmd('FileType', {
        pattern = 'java',
        group = augroup,
        callback = setup_jdtls,
      })
    end,
  },
})
```

:Telescope lsp_definitions
Error: [telescope.builtin.lsp_*]: server does not support definitionProvider

### Expected behavior

the same behaviour as with vim.lsp.buf.definition() (which does work)

### Actual behavior

I get the errorr: [telescope.builtin.lsp_*]: server does not support definitionProvider

### Minimal config

```Lua
-- Fetches Lazy if not present.
local function bootstrap_lazy()
  local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
  if not vim.loop.fs_stat(lazypath) then
    vim.fn.system({
      "git",
      "clone",
      "--filter=blob:none",
      "https://github.com/folke/lazy.nvim.git",
      "--branch=stable",
      lazypath,
    })
  end
  vim.opt.rtp:prepend(lazypath)
end
bootstrap_lazy()

require('lazy').setup({
  {
    'williamboman/mason.nvim',
    opts = {},
  },
  {
    'mfussenegger/nvim-jdtls',
    dependencies = {
      'williamboman/mason.nvim',
      'neovim/nvim-lspconfig',
      { 'nvim-telescope/telescope.nvim', dependencies = 'nvim-lua/plenary.nvim' },
    },
    config = function()
      local function setup_jdtls()
        local mason_path = require('mason-core.path')
        local function get_install_path(package)
          return require('mason-registry').get_package(package):get_install_path()
        end

        local config = {
          cmd = { mason_path.concat({ get_install_path('jdtls'), 'jdtls' }) },
        }
        require('jdtls').start_or_attach(config)
      end

      local augroup = vim.api.nvim_create_augroup('jdtls', {})
      vim.api.nvim_create_autocmd('FileType', {
        pattern = 'java',
        group = augroup,
        callback = setup_jdtls,
      })
    end,
  },
})
```

**Comment by jamestrew** [comment]
Hmm.. I can't reproduce this using your provided config + minimal config
```lua
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  {
    "williamboman/mason.nvim",
    opts = {},
  },
  {
    "mfussenegger/nvim-jdtls",
    dependencies = {
      "williamboman/mason.nvim",
      "neovim/nvim-lspconfig",
      { "nvim-telescope/telescope.nvim", dependencies = "nvim-lua/plenary.nvim" },
    },
    config = function()
      local function setup_jdtls()
        local mason_path = require("mason-core.path")
        local function get_install_path(package)
          return require("mason-registry").get_package(package):get_install_path()
        end

        local config = {
          cmd = { mason_path.concat({ get_install_path("jdtls"), "jdtls" }) },
        }
        require("jdtls").start_or_attach(config)
      end

      local augroup = vim.api.nvim_create_augroup("jdtls", {})
      vim.api.nvim_create_autocmd("FileType", {
        pattern = "java",
        group = augroup,
        callback = setup_jdtls,
      })
    end,
  },
  {
    "nvim-telescope/telescope.nvim",
    dependencies = {
      "nvim-lua/plenary.nvim",
      "nvim-tree/nvim-web-devicons",
    },
    config = function()
      require("telescope").setup({})
      vim.keymap.set("n", "<C-p>", ":Telescope find_files<CR>")
    end,
  },
}

require("lazy").setup(plugins, {
  root = root .. "/plugins",
})
```

https://github.com/nvim-telescope/telescope.nvim/assets/66286082/f9327e83-b22b-40ef-9a5d-c5edf578bf09

The suggest in the original issue https://github.com/mfussenegger/nvim-jdtls/issues/589 
https://github.com/mfussenegger/nvim-jdtls/issues/589#issuecomment-1821361157
this is something we already do.

https://github.com/nvim-telescope/telescope.nvim/blob/e54fbf4ab2b64640b639b75c006c23b4bc51c86f/lua/telescope/builtin/__lsp.lua#L408-L456

**Comment by fpbouchard** [comment]
I have found the source of the issue which is that jdtls uses dynamic registration for some supported server capabilities. In neovim nightly, this is now supported but there is a change in the API to request server capabilities support: 

https://github.com/neovim/neovim/blob/b465ede2c7a4fb39cf84682d645a3acd08631010/runtime/doc/news.txt#L220

jBy changing https://github.com/nvim-telescope/telescope.nvim/blob/e54fbf4ab2b64640b639b75c006c23b4bc51c86f/lua/telescope/builtin/__lsp.lua#L412 to use the new API, it fixes the issue for me.

**Comment by fpbouchard** [comment]
@fgheng I had the same issue, but it is indeed fixed in the master branch. The code that @jamestrew refers to is using `supports_method` which is the new API that is in nvim 0.10 (nightly) that supports dynamic server capability registration. When unlocking my telescope version and using nightly, this works.

**Comment by nithin-mk** [comment]
Now that NeoVim 0.10.0 is the latest stable release, please can we have a new release of telescope which includes this fix?

**Comment by Juniar-Rakhman** [comment]
can anyone confirm if the latest version (0.1.7) contains this fix?

**Comment by Richie-Z** [comment]
> can anyone confirm if the latest version (0.1.7) contains this fix?

in dartls it still error, I don't know in nvim-jdtls

<img width="1470" alt="Screenshot 2024-05-22 at 09 36 09" src="https://github.com/nvim-telescope/telescope.nvim/assets/48403322/68c8f2a8-91e9-4027-b17b-d70b160ae985">

**Comment by ltfourrier** [comment]
No, the fix is not part of 0.1.7. Quoting jamestrew from [the PR that fixes this](https://github.com/nvim-telescope/telescope.nvim/pull/2594):

> we aren't cherry picking features into 0.1.x following semver. This will come in 0.2.0 but no timeline for this. Best bet is using the master branch pinned to this commit.

**Comment by jeeeem** [comment]
> > can anyone confirm if the latest version (0.1.7) contains this fix?
> 
> in dartls it still error, I don't know in nvim-jdtls
It still has the error

**Comment by Conni2461** [comment]
we need to find a solution for this. problem is, that 0.1. still supports nvim 0.7 so yeah. maybe thats not actually an features but more of a fix.

I'll will handle this and probably release 0.1.8 later this evening. sorry for the inconvenience

**Comment by jeeeem** [comment]
> the backported patch is pushed to 0.1.x, i dont wanna setup jdtls right now. can someone confirm that the issue is fixed on that branch, if yes i'll do a 0.1.8 release.
> 
> this should also not even be a breaking change, because `client.supports_method` is a really old thing [neovim/neovim#12764](https://github.com/neovim/neovim/pull/12764).
> 
> backport: [08a2810](https://github.com/nvim-telescope/telescope.nvim/commit/08a28108c23323ed1f5c1a68b34bfdb2590fe98b)
> 
> sorry for the inconvenience

I just tried it on JDTLS, and it works properly well. 
![image](https://github.com/nvim-telescope/telescope.nvim/assets/51053482/16fa3a43-e4dc-4919-ae0d-2e4d430fbb7a)

Other infos:
NVIM v0.10.0
Astronvim v4.7.5

**Comment by Conni2461** [comment]
thanks for testing, is now released with 0.1.8: https://github.com/nvim-telescope/telescope.nvim/releases/tag/0.1.8
